Missing type-utils.d.ts file in @middy/util package

**Describe the bug**
When importing @middy/util in a TypeScript project, tsc fails with the following error

```
node_modules/@middy/util/index.d.ts:3:88 - error TS2834: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

3 import { ArrayValues, Choose, DeepAwaited, IsUnknown, SanitizeKey, SanitizeKeys } from './type-utils'
```

if I correct the import by adding the .js extension, then tsc fails with another error

```
node_modules/@middy/util/index.d.ts:3:88 - error TS2307: Cannot find module './type-utils.js' or its corresponding type declarations.

3 import { ArrayValues, Choose, DeepAwaited, IsUnknown, SanitizeKey, SanitizeKeys } from './type-utils.js'
```

In fact, the type-utils.d.ts file is not present.

**To Reproduce**
How to reproduce the behaviour:

See sample at https://github.com/lbevilacqua/middy-util-issue

**Expected behaviour**
tsc command should not produce the above errors.

**Environment (please complete the following information):**

- Node.js: 18.18.0
- Middy: 5.0.0

**Additional context**
package.json type: module
tsconfig.json module: node16

--- a/packages/util/index.d.ts
+++ b/packages/util/index.d.ts
@@ -1,6 +1,13 @@
 import middy from '@middy/core'
 import { Context as LambdaContext } from 'aws-lambda'
-import { ArrayValues, Choose, DeepAwaited, IsUnknown, SanitizeKey, SanitizeKeys } from './type-utils'
+import type {
+  ArrayValues,
+  Choose,
+  DeepAwaited,
+  IsUnknown,
+  SanitizeKey,
+  SanitizeKeys
+} from './type-utils.d.ts'
 
 interface Options<Client, ClientOptions> {
   AwsClient?: new (...[config]: [any] | any) => Client
@@ -17,31 +24,33 @@ interface Options<Client, ClientOptions> {
 declare class HttpError extends Error {
   status: number
   statusCode: number
-  expose: boolean
+  expose: boolean;
   [key: string]: any
   [key: number]: any
 }
 
-declare function createPrefetchClient<Client, ClientOptions> (
+declare function createPrefetchClient<Client, ClientOptions>(
   options: Options<Client, ClientOptions>
 ): Client
 
-declare function createClient<Client, ClientOptions> (
+declare function createClient<Client, ClientOptions>(
   options: Options<Client, ClientOptions>,
   request: middy.Request
 ): Promise<Client>
 
-declare function canPrefetch<Client, ClientOptions> (
+declare function canPrefetch<Client, ClientOptions>(
   options: Options<Client, ClientOptions>
 ): boolean
 
-type InternalOutput<TVariables> = TVariables extends string[] ? { [key in TVariables[number]]: unknown } : never
+type InternalOutput<TVariables> = TVariables extends string[]
+  ? { [key in TVariables[number]]: unknown }
+  : never
 
 // get an empty object if false is passed
 declare function getInternal<
   TContext extends LambdaContext,
   TInternal extends Record<string, unknown>
-> (
+>(
   variables: false,
   request: middy.Request<unknown, unknown, unknown, TContext, TInternal>
 ): Promise<{}>
@@ -50,7 +59,7 @@ declare function getInternal<
 declare function getInternal<
   TContext extends LambdaContext,
   TInternal extends Record<string, unknown>
-> (
+>(
   variables: true,
   request: middy.Request<unknown, unknown, unknown, TContext, TInternal>
 ): Promise<DeepAwaited<TInternal>>
@@ -60,77 +69,79 @@ declare function getInternal<
   TContext extends LambdaContext,
   TInternal extends Record<string, unknown>,
   TVars extends keyof TInternal | string
-> (
+>(
   variables: TVars,
   request: middy.Request<unknown, unknown, unknown, TContext, TInternal>
 ): TVars extends keyof TInternal
   ? Promise<DeepAwaited<{ [_ in SanitizeKey<TVars>]: TInternal[TVars] }>>
   : TVars extends string
-    ? IsUnknown<Choose<DeepAwaited<TInternal>, TVars>> extends true
-      ? unknown // could not find the path
-      : Promise<{ [_ in SanitizeKey<TVars>]: Choose<DeepAwaited<TInternal>, TVars> }>
-    : unknown // path is not a string or a keyof TInternal
+  ? IsUnknown<Choose<DeepAwaited<TInternal>, TVars>> extends true
+    ? unknown // could not find the path
+    : Promise<{
+        [_ in SanitizeKey<TVars>]: Choose<DeepAwaited<TInternal>, TVars>
+      }>
+  : unknown // path is not a string or a keyof TInternal
 
 // get multiple values
 declare function getInternal<
   TContext extends LambdaContext,
   TInternal extends Record<string, unknown>,
   TVars extends Array<keyof TInternal | string>
-> (
+>(
   variables: TVars,
   request: middy.Request<unknown, unknown, unknown, TContext, TInternal>
-): Promise<SanitizeKeys<{
-  [TVar in ArrayValues<TVars>]:
-  TVar extends keyof TInternal
-    ? DeepAwaited<TInternal[TVar]>
-    : TVar extends string
+): Promise<
+  SanitizeKeys<{
+    [TVar in ArrayValues<TVars>]: TVar extends keyof TInternal
+      ? DeepAwaited<TInternal[TVar]>
+      : TVar extends string
       ? Choose<DeepAwaited<TInternal>, TVar>
       : unknown // path is not a string or a keyof TInternal
-}>>
+  }>
+>
 
 // remap object
 declare function getInternal<
   TContext extends LambdaContext,
   TInternal extends Record<string, unknown>,
   TMap extends Record<string, keyof TInternal | string>
-> (
+>(
   variables: TMap,
   request: middy.Request<unknown, unknown, unknown, TContext, TInternal>
 ): Promise<{
-  [P in keyof TMap]:
-  TMap[P] extends keyof TInternal
+  [P in keyof TMap]: TMap[P] extends keyof TInternal
     ? DeepAwaited<TInternal[TMap[P]]>
     : TMap[P] extends string
-      ? Choose<DeepAwaited<TInternal>, TMap[P]>
-      : unknown // path is not a string or a keyof TInternal
+    ? Choose<DeepAwaited<TInternal>, TMap[P]>
+    : unknown // path is not a string or a keyof TInternal
 }>
 
-declare function sanitizeKey<T extends string> (key: T): SanitizeKey<T>
+declare function sanitizeKey<T extends string>(key: T): SanitizeKey<T>
 
-declare function processCache<Client, ClientOptions> (
+declare function processCache<Client, ClientOptions>(
   options: Options<Client, ClientOptions>,
   fetch: (request: middy.Request, cachedValues: any) => any,
   request?: middy.Request
-): { value: any, expiry: number }
+): { value: any; expiry: number }
 
-declare function getCache (keys: string): any
+declare function getCache(keys: string): any
 
-declare function clearCache (keys?: string | string[] | null): void
+declare function clearCache(keys?: string | string[] | null): void
 
-declare function jsonSafeParse (
+declare function jsonSafeParse(
   string: string,
   reviver?: (key: string, value: any) => any
 ): any
 
-declare function normalizeHttpResponse (
+declare function normalizeHttpResponse(
   request: any,
   fallbackResponse?: any
 ): any
 
-declare function createError (
+declare function createError(
   code: number,
   message: string,
   properties?: Record<string, any>
 ): HttpError
 
-declare function modifyCache (cacheKey: string, value: any): void
+declare function modifyCache(cacheKey: string, value: any): void

--- a/packages/util/type-utils.test-d.ts
+++ b/packages/util/type-utils.test-d.ts
@@ -1,5 +1,5 @@
 import { expectType } from 'tsd'
-import {
+import type {
   SanitizeKeyPrefixLeadingNumber,
   SanitizeKeyRemoveDisallowedChar,
   RemoveAllLeadingUnderscore,
@@ -10,7 +10,7 @@ import {
   IsUnknown,
   ArrayValues,
   SanitizeKeys
-} from './type-utils'
+} from './type-utils.d.ts'
 
 // SanitizeKeyPrefixLeadingNumber
 // adds a leading _ if it starts with a number
@@ -27,7 +27,9 @@ expectType<SanitizeKeyPrefixLeadingNumber<'-abcd'>>('-abcd')
 
 // SanitizeKeyRemoveDisallowedChar
 // removes all disallowed chars and replaces them with _
-expectType<SanitizeKeyRemoveDisallowedChar<'1234!@#AA$%^&*()BBB_+{}|:"<>?~CC`-=[D]E\\;,./F'>>('1234___AA_______BBB___________CC____D_E_____F')
+expectType<
+  SanitizeKeyRemoveDisallowedChar<'1234!@#AA$%^&*()BBB_+{}|:"<>?~CC`-=[D]E\\;,./F'>
+>('1234___AA_______BBB___________CC____D_E_____F')
 
 // RemoveAllLeadingUnderscore
 // removes all leading _
@@ -57,11 +59,20 @@ expectType<SanitizeKey<'0key.0key'>>('_0key_0key')
 // SanitizeKeys
 // sanitizes all keys in an object
 expectType<SanitizeKeys<{ '0key': 0 }>>({ _0key: 0 })
-expectType<SanitizeKeys<{ 'api//secret-key0.pem': 0 }>>({ api_secret_key0_pem: 0 })
-expectType<SanitizeKeys<{ '0key': 0, 'api//secret-key0.pem': 0 }>>({ _0key: 0, api_secret_key0_pem: 0 })
+expectType<SanitizeKeys<{ 'api//secret-key0.pem': 0 }>>({
+  api_secret_key0_pem: 0
+})
+expectType<SanitizeKeys<{ '0key': 0; 'api//secret-key0.pem': 0 }>>({
+  _0key: 0,
+  api_secret_key0_pem: 0
+})
 
 // DeepAwaited
-expectType<DeepAwaited<{ level1: { level2: Promise<Promise<{ innerPromise: Promise<22> }>> } }>>({
+expectType<
+  DeepAwaited<{
+    level1: { level2: Promise<Promise<{ innerPromise: Promise<22> }>> }
+  }>
+>({
   level1: {
     level2: {
       innerPromise: Promise.resolve(22) // Note: getInternal does not recurse into resolved promises
@@ -88,7 +99,9 @@ expectType<Choose<TestChoose, 'number'>>(1)
 expectType<Choose<TestChoose, 'object.key'>>('value')
 expectType<Choose<TestChoose, 'promise'>>(Promise.resolve('string'))
 expectType<Choose<DeepAwaited<TestChoose>, 'promise'>>('string')
-expectType<Choose<TestChoose, 'promiseObject'>>(Promise.resolve({ key: 'value' }))
+expectType<Choose<TestChoose, 'promiseObject'>>(
+  Promise.resolve({ key: 'value' })
+)
 expectType<Choose<DeepAwaited<TestChoose>, 'promiseObject'>>({ key: 'value' })
 
 // IsUnknown

--- a/packages/validator/transpile.test-d.ts
+++ b/packages/validator/transpile.test-d.ts
@@ -1,5 +1,5 @@
 import { expectType } from 'tsd'
-import { transpileSchema, transpileLocale } from './transpile'
+import type { transpileSchema, transpileLocale } from './transpile.t.ds'
 
 const schema = transpileSchema({ type: 'object' }, {})
 expectType<any>(schema)

